controllers/openshift_controller.go

```diff
@@ -773,6 +773,17 @@ func (r *KataConfigOpenShiftReconciler) processKataConfigInstallRequest() (ctrl.
 			return ctrl.Result{}, err
 		}
 
+		// Update node selector in machine config pool with value from kataconfig instance
+		r.Log.Info("Updating machine config pool")
+		if foundMcp != nil {
+			*foundMcp.Spec.NodeSelector = *r.kataConfig.Spec.KataConfigPoolSelector
+			err = r.Client.Update(context.TODO(), foundMcp)
```

Comment:
Interesting approach to handle the case of kataconfig update post installation.
I'm wondering if it would make sense to watch for changes to kataconfig in general and take care of updating mutable fields in kataconfig ? Or it's over kill at this moment ?

Reply:
@bpradipt I'd say if a KataConfig field is mutable, in principle we should handle its change. Otherwise we're bound to have clients changing fields and finding out they do nothing - not very good long-terms... :-)

@jensfr Would it make sense to check if the selector has _actually_ changed before calling `r.Client.Update()`? Updating an MCP isn't a lightweight operation and unless updating an MCP without actually changing it is optimised (=ignored) by MCO itself we might consider doing it on our side I suppose.

Reply:
@pmores valid point. Just that I'm not sure if we want all the fields to be mutable and the related consequences.
Regarding your question on checking the selector, afaik it's not needed as the update operation will be a no-op if there are no changes to the object.

Reply:
I was going for the smallest change possible for this fix, but I agree that we should take a closer look at what other changes in kataconfig we should act on. 

Good point @pmores, I was counting on MCO not doing anything if there are no changes in the applied config. Similar to how it says 'no changes' when you apply the same yaml twice from the command-line.